ansible_base/authentication/serializers/authenticator_map.py

```diff
@@ -17,17 +25,69 @@ def validate(self, data) -> dict:
         map_type = data.get('map_type', None)
         team = data.get('team', None)
         org = data.get('organization', None)
-        if map_type == 'team' and (not team or team == ''):
-            errors["team"] = "You must specify a team with the selected map type"
-        if map_type == 'team' and (not org or org == ''):
-            errors["organization"] = "You must specify an organization with the selected map type"
-        if map_type == 'organization' and (not org or org == ''):
-            errors["organization"] = "You must specify an organization with the selected map type"
+        role = data.get('role', None)
+
+        if map_type == 'team' and not team:
+            errors["team"] = _("You must specify a team with the selected map type")
+        if map_type in ['team', 'organization'] and not org:
+            errors["organization"] = _("You must specify an organization with the selected map type")
+        if map_type in ['team', 'organization', 'role'] and not role:
+            errors["role"] = _("You must specify a role with the selected map type")
+        if map_type in ['allow', 'is_superuser'] and role:
+            errors["role"] = _("You cannot specify role with the selected map type")
+
+        if role:
+            errors.update(self.validate_role_data(map_type, role, org, team))
 
         if errors:
             raise ValidationError(errors)
         return data
 
+    def validate_role_data(self, map_type, role, org, team):
+        errors = {}
+
+        # Validation is possible only if RBAC is installed
+        if not apps.is_installed('ansible_base.rbac'):
+            logger.warning(_("You specified a role without RBAC installed "))
+            return errors
+
+        from ansible_base.rbac.models import RoleDefinition
+
+        try:
+            rbac_role = RoleDefinition.objects.get(name=role)
+            is_system_role = rbac_role.content_type is None
+
+            # system role is allowed for map type == role without further conditions
+            if is_system_role and map_type == 'role':
+                return errors
+
+            if is_system_role:
+                is_org_role, is_team_role = False, False
+            else:
+                model_class = rbac_role.content_type.model_class()
+                is_org_role = issubclass(model_class, get_organization_model())
+                is_team_role = issubclass(model_class, get_team_model())
```

Comment:
We should be able to get rid of this if/else?

```suggestion
            is_org_role, is_team_role = False, False
            model_class = rbac_role.content_type.model_class()
            is_org_role = issubclass(model_class, get_organization_model())
            is_team_role = issubclass(model_class, get_team_model())
```

Reply:
we can't, because in case of system_role the content_type is None, which means `.model_class` will be called on None

Reply:
Ah, good call.